lib/shelley/src/Cardano/Wallet/Shelley/Network/Blockfrost.hs

```diff
@@ -236,6 +245,22 @@ withNetworkLayer tr net np project k = k NetworkLayer
         mkTimeInterpreter (MsgTimeInterpreterLog >$< tr) startTime $
             pure $ HF.mkInterpreter $ networkSummary net
 
+    fetchRewardAccountBalances ::
+        Set RewardAccount -> IO (Map RewardAccount Coin)
+    fetchRewardAccountBalances accounts =
+        handleBlockfrostError . fmap Map.fromList $
+            for (Set.toList accounts) $ \rewardAccount -> do
+                BF.AccountInfo {..} <- liftBlockfrost $ BF.getAccount $
+                    BF.mkAddress $ toText rewardAccount
```

Comment:
Unfortunately, I that think `toText` is not correct here 🤔 — the `RewardAccount` type stores a raw Bytestring while Blockfrost expects a Bech32-encoded text (of that raw Bytestring). I don't quite know what `toText` does, but at best, it will only give a hex-encoding — which is not what Blockfrost expects.

I think that the `encodeStakeAddress` function is relevant here. Unfortunately, I don't know the details on top of my head. [CIP-19](https://cips.cardano.org/cips/cip19/) contains general information on encodings and address formats.